addons/hr_recruitment/hr_recruitment.py

```diff
@@ -309,9 +302,7 @@ def stage_find(self, cr, uid, applicants, job_id, domain=[], order='sequence', c
             job_ids.add(job_id)
         # OR all job_id and OR with applicant_default
         if job_ids:
-            search_domain = ['|', ('case_default', '=', True), ('job_ids', 'in', list(job_ids))]
-        else:
-            search_domain = [('case_default', '=', True)]
+            search_domain = [('job_ids', 'in', list(job_ids))]
```

Comment:
Don't bother fixing it, I'll fix it myself when merging as the rest of the code is ok, it was just FYI.